### PR TITLE
Adds the possiblity to force a plugin installation - elasticsearch_plugin module

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY
Some elasticsearch plugins needs an additional confirmation for further permissions in the java. So it's not possible to install these plugins without the --batch mode in the elasticsearch plugin installation process. This mode ignores the confirmation prompt.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
elasticsearch_plugin

##### ANSIBLE VERSION

```
ansible 2.5.5
  config file = None
  configured module search path = [u'~/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Python/2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.10 (default, Oct  6 2017, 22:29:07) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### ADDITIONAL INFORMATION

Before:
```TASK [elasticsearch : install elasticsearch plugins] ********************************************************************************************************************************************************************************************************************
failed: [node01] (item={u'state': u'present', u'name': u'ingest-geoip'}) => {"changed": false, "err": "-> Downloading ingest-geoip from elastic\n\r[                                                 ] 1%   \r[>                                                ] 2%   \r[>                                                ] 3%   \r[=>                                               ] 4%   \r[=>                                               ] 5%   \r[==>                                              ] 6%   \r[==>                                              ] 7%   \r[===>                                             ] 8%   \r[===>                                             ] 9%   \r[====>                                            ] 10%   \r[====>                                            ] 11%   \r[=====>                                           ] 12%   \r[=====>                                           ] 13%   \r[======>                                          ] 14%   \r[======>                                          ] 15%   \r[=======>                                         ] 16%   \r[=======>                                         ] 17%   \r[========>                                        ] 18%   \r[========>                                        ] 19%   \r[=========>                                       ] 20%   \r[=========>                                       ] 21%   \r[==========>                                      ] 22%   \r[==========>                                      ] 23%   \r[===========>                                     ] 24%   \r[===========>                                     ] 25%   \r[============>                                    ] 26%   \r[============>                                    ] 27%   \r[=============>                                   ] 28%   \r[=============>                                   ] 29%   \r[==============>                                  ] 30%   \r[==============>                                  ] 31%   \r[===============>                                 ] 32%   \r[===============>                                 ] 33%   \r[================>                                ] 34%   \r[================>                                ] 35%   \r[=================>                               ] 36%   \r[=================>                               ] 37%   \r[==================>                              ] 38%   \r[==================>                              ] 39%   \r[===================>                             ] 40%   \r[===================>                             ] 41%   \r[====================>                            ] 42%   \r[====================>                            ] 43%   \r[=====================>                           ] 44%   \r[=====================>                           ] 45%   \r[======================>                          ] 46%   \r[======================>                          ] 47%   \r[=======================>                         ] 48%   \r[=======================>                         ] 49%   \r[========================>                        ] 50%   \r[========================>                        ] 51%   \r[=========================>                       ] 52%   \r[=========================>                       ] 53%   \r[==========================>                      ] 54%   \r[==========================>                      ] 55%   \r[===========================>                     ] 56%   \r[===========================>                     ] 57%   \r[============================>                    ] 58%   \r[============================>                    ] 59%   \r[=============================>                   ] 60%   \r[=============================>                   ] 61%   \r[==============================>                  ] 62%   \r[==============================>                  ] 63%   \r[===============================>                 ] 64%   \r[===============================>                 ] 65%   \r[================================>                ] 66%   \r[================================>                ] 67%   \r[=================================>               ] 68%   \r[=================================>               ] 69%   \r[==================================>              ] 70%   \r[==================================>              ] 71%   \r[===================================>             ] 72%   \r[===================================>             ] 73%   \r[====================================>            ] 74%   \r[====================================>            ] 75%   \r[=====================================>           ] 76%   \r[=====================================>           ] 77%   \r[======================================>          ] 78%   \r[======================================>          ] 79%   \r[=======================================>         ] 80%   \r[=======================================>         ] 81%   \r[========================================>        ] 82%   \r[========================================>        ] 83%   \r[=========================================>       ] 84%   \r[=========================================>       ] 85%   \r[==========================================>      ] 86%   \r[==========================================>      ] 87%   \r[===========================================>     ] 88%   \r[===========================================>     ] 89%   \r[============================================>    ] 90%   \r[============================================>    ] 91%   \r[=============================================>   ] 92%   \r[=============================================>   ] 93%   \r[==============================================>  ] 94%   \r[==============================================>  ] 95%   \r[===============================================> ] 96%   \r[===============================================> ] 97%   \r[================================================>] 98%   \r[================================================>] 99%   \r[=================================================] 100%   \n@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n@     WARNING: plugin requires additional permissions     @\n@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n* java.lang.RuntimePermission accessDeclaredMembers\n* java.lang.reflect.ReflectPermission suppressAccessChecks\nSee http://docs.oracle.com/javase/8/docs/technotes/guides/security/permissions.html\nfor descriptions of what these permissions allow and the associated risks.\n\n", "item": {"name": "ingest-geoip", "state": "present"}, "msg": "Is ingest-geoip a valid plugin name?"}
```
After:
```
TASK [elasticsearch : install elasticsearch plugins] ********************************************************************************************************************************************************************************************************************
changed: [node01] => (item={u'state': u'present', u'name': u'ingest-geoip'})
```

### CODE REVIEW

File: lib/ansible/modules/database/misc/elasticsearch_plugin.py [hunk 1]
```diff
@@ -43,6 +43,11 @@
             - "Timeout setting: 30s, 1m, 1h..."
             - Only valid for Elasticsearch < 5.0. This option is ignored for Elasticsearch > 5.0.
         default: 1m
+    force:
+        description:
+            - "Give the possibility to force a plugin installation even with addition permission needed"
```
Comment (samdoran) [inline lib/ansible/modules/database/misc/elasticsearch_plugin.py]: I would change the wording on this to align more with the [Elatic docs](https://www.elastic.co/guide/en/elasticsearch/plugins/6.3/_other_command_line_parameters.html#_batch_mode) and indicate what this parameter is doing under the hood. Something along the lines of "Force batch mode when installing plugins. This is only necessary if a plugin requires additional permissions and console detection fails."

**Review by samdoran — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/database/misc/elasticsearch_plugin.py (hunk 1 above)
Comment (ghost) [reply lib/ansible/modules/database/misc/elasticsearch_plugin.py]: Good idea. I've changed this.

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
Looks good! Please create a changelog fragment and I'll merge this. See [fragments](https://github.com/ansible/ansible/tree/devel/changelogs/fragments) for examples.

**Comment by ghost** [comment]
@samdoran Done! ;)

File: lib/ansible/modules/database/misc/elasticsearch_plugin.py [hunk 2]
```diff
@@ -85,6 +90,12 @@
 - elasticsearch_plugin:
     name: analysis-icu
     state: present
+
+# Install a the ingest-geoip plugin with a forced installation
```
Comment (samdoran) [inline lib/ansible/modules/database/misc/elasticsearch_plugin.py]: Remove the 'a'.

File: changelogs/fragments/batch_mode_elasticsearch_plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - Add the possibility to use the elasticsearch_plugin installation batch mode
```
Comment (samdoran) [inline changelogs/fragments/batch_mode_elasticsearch_plugin.yaml]: Start this entry with `- elasticsearch_plugin - `. This make the module name easier to find when visually scanning the changelogs.

**Review by samdoran — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@samdoran done again! ;)

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
@Nitaco Thank you for all the timely fixes and for your contribution!

**Pull request merged**